sale_advance_payment/wizard/sale_advance_payment_wzd.py

```diff
@@ -0,0 +1,145 @@
+    )
+    currency_id = fields.Many2one("res.currency", "Currency", readonly=True)
+    amount_total = fields.Monetary("Amount total", readonly=True)
+    amount_advance = fields.Monetary("Amount advanced", required=True)
+    date = fields.Date("Date", required=True, default=fields.Date.context_today)
+    exchange_rate = fields.Float(
+        "Exchange rate", digits=(16, 6), default=1.0, readonly=True
+    )
+    currency_amount = fields.Monetary(
+        "Curr. amount", readonly=True, currency_field="journal_currency_id"
+    )
+    payment_ref = fields.Char("Ref.")
+
+    @api.constrains("amount_advance")
+    def check_amount(self):
+        if self.amount_advance <= 0:
+            raise exceptions.ValidationError(_("Amount of advance must be positive."))
+        if self.env.context.get("active_id", False):
+            order = self.env["sale.order"].browse(self.env.context["active_id"])
+            if self.amount_advance > order.amount_residual:
+                raise exceptions.ValidationError(
+                    _("Amount of advance is greater than residual amount on sale")
+                )
+
+    @api.model
+    def default_get(self, fields_list):
+        res = super().default_get(fields_list)
+        sale_ids = self.env.context.get("active_ids", [])
+        if not sale_ids:
+            return res
+        sale_id = sale_ids[0]
+
+        sale = self.env["sale.order"].browse(sale_id)
+
+        if "amount_total" in fields_list:
+            res.update(
+                {
+                    "amount_total": sale.amount_residual,
+                    "currency_id": sale.pricelist_id.currency_id.id,
+                }
+            )
+
+        return res
+
+    @api.onchange("journal_id", "date", "amount_advance")
+    def onchange_date(self):
+
+        sale_ids = self.env.context.get("active_ids", [])
+        sale_obj = self.env["sale.order"]
+        if sale_ids:
+            sale_id = sale_ids[0]
+            sale = sale_obj.browse(sale_id)
+            if self.currency_id:
+                self.exchange_rate = 1.0 / (
+                    self.env["res.currency"]._get_conversion_rate(
+                        sale.company_id.currency_id,
+                        self.currency_id,
+                        sale.company_id,
+                        self.date,
+                    )
+                    or 1.0
+                )
+            else:
+                self.exchange_rate = 1.0
+            self.currency_amount = self.amount_advance * (1.0 / self.exchange_rate)
+
+    def make_advance_payment(self):
+        """Create customer paylines and validates the payment"""
+        self.ensure_one()
+        payment_obj = self.env["account.payment"]
+        sale_obj = self.env["sale.order"]
+
+        sale_ids = self.env.context.get("active_ids", [])
+        if sale_ids:
+            sale_id = sale_ids[0]
+            sale = sale_obj.browse(sale_id)
+
+            partner_id = sale.partner_id.commercial_partner_id.id
```

Comment:
I'm not sure that's correct for standard flow. Should be at least partner_invoice_id no ?

Reply:
@rousseldenis You were right! Thank you for the review!